src/Coordination/KeeperSnapshotManagerS3.cpp

```diff
@@ -64,7 +65,8 @@ void KeeperSnapshotManagerS3::updateS3Configuration(const Poco::Util::AbstractCo
 
         auto auth_settings = S3::AuthSettings::loadFromConfig(config_prefix, config);
 
-        auto endpoint = config.getString(config_prefix + ".endpoint");
+        String endpoint = config.getString(config_prefix + ".endpoint");
+        endpoint = macros->expand(endpoint);
```

Comment:
Why not do it in a single line? 
`macros->expand(config.getString(config_prefix + ".endpoint"))`

Reply:
good point, addressed it